cranelift/codegen/src/egraph.rs

```diff
@@ -266,7 +312,7 @@ where
         // Create a union of all new values with the original (or
         // maybe just one new value marked as "subsuming" the
         // original, if present.)
-        let mut union_value = orig_value;
+        let mut union_value = optimized_values.pop().unwrap();
```

Comment:
Rather than the indirect pushing and then popping, this could be

```suggestion
        let mut union_value = optimized_values.pop().unwrap_or(orig_value);
```

Then, as a reader, I don't have to wonder if the push is going to trigger a realloc or not (probably not because it is a smallvec and we limit the number of returns from ISLE in ISLE, but I think it is possible if we return the max ISLE returns and then push `orig_vlaue`?)